iris_grib/tests/unit/save_rules/test_set_fixed_surfaces.py

```diff
@@ -118,10 +120,89 @@ def test_height_point(self, mock_set):
+        message = "vertical-axis coordinate unit may not be encoded correctly"
+        self.assertRegex(str(warn[0].message), message)
+
+        self.assertEqual(gribapi.grib_get_long(
+            grib, "typeOfFirstFixedSurface"), 151)
+        self.assertEqual(gribapi.grib_get_double(
+            grib, "scaledValueOfFirstFixedSurface"), 1)
+        self.assertEqual(gribapi.grib_get_double(
+            grib, "scaleFactorOfFirstFixedSurface"), 0)
+        self.assertEqual(gribapi.grib_get_long(
+            grib, "typeOfSecondFixedSurface"), 255)
+        self.assertEqual(gribapi.grib_get_long(
+            grib, "scaledValueOfSecondFixedSurface"), GRIB_MISSING_LONG)
+        self.assertEqual(gribapi.grib_get_long(
+            grib, "scaleFactorOfSecondFixedSurface"), GRIB_MISSING_LONG)
+
+    def test_unknown_vertical_bounded(self):
+        cube = iris.cube.Cube([0])
+        cube.add_aux_coord(
+            iris.coords.AuxCoord([450], bounds=np.array([900.0, 0.0]),
+                                 attributes={'GRIB_fixed_surface_type': 108}))
+        grib = gribapi.grib_new_from_samples("GRIB2")
+        with warnings.catch_warnings(record=True) as warn:
+            set_fixed_surfaces(cube, grib)
+
+        self.assertEqual(len(warn), 1)
+        message = "vertical-axis coordinate unit may not be encoded correctly"
+        self.assertRegex(str(warn[0].message), message)
+
+        self.assertEqual(
+            gribapi.grib_get_long(grib, "typeOfFirstFixedSurface"), 108)
+        self.assertEqual(
+            gribapi.grib_get_double(grib, "scaledValueOfFirstFixedSurface"),
+            900)
+        self.assertEqual(
+            gribapi.grib_get_double(grib, "scaleFactorOfFirstFixedSurface"),
+            0)
+        self.assertEqual(
+            gribapi.grib_get_long(grib, "typeOfSecondFixedSurface"), 108)
+        self.assertEqual(
+            gribapi.grib_get_long(grib, "scaledValueOfSecondFixedSurface"),
+            0)
+        self.assertEqual(
+            gribapi.grib_get_long(grib, "scaleFactorOfSecondFixedSurface"),
+            0)
+
+    def test_multiple_unknown_vertical_coords(self):
+        grib = None
+        cube = iris.cube.Cube([0])
+        cube.add_aux_coord(
+            iris.coords.AuxCoord([1],
+                                 attributes={'GRIB_fixed_surface_type': 151}))
+        cube.add_aux_coord(
+            iris.coords.AuxCoord([450], bounds=np.array([900.0, 0.0]),
+                                 attributes={'GRIB_fixed_surface_type': 108}))
+        msg = r"coordinates were found of fixed surface type: \[151, 108\]"
+        with self.assertRaisesRegex(ValueError, msg):
+            set_fixed_surfaces(cube, grib)
+
+    def test_unhandled_vertical_axis(self):
+        grib = None
+        cube = iris.cube.Cube([0])
+        cube.add_aux_coord(
+            iris.coords.AuxCoord([450], attributes={'positive': 'up'}))
+        msg = r"coordinate\(s\) \('unknown'\) are not recognised or handled."
```

Comment:
I feel it might be worthwhile extending the scope of the message match here, just for a bit more clarity.
I.E. at least as far as to say  "vertical-axis coordinate(s) ...".